From 2830eab44753bfce6bed99a7c8e13e742fa2c3eb Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Fri, 7 May 2021 09:10:50 -0700
Subject: [PATCH 1/7] octeontx2-af: Increase the NIX LF resources for 98xx

commit 8280b9092e51f9fc4574ca65c975fa2ff06073d2 from
git@git.assembla.com:cavium/WindRiver.linux.git

98xx has NIX1, so increase the total limit by adding NIX1 LFs as well.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: I6b4fea3910b548395e536a2e30c5d7a12d96a0ca
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/51413
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
index 3315ad414386..8c42b0951e0a 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
@@ -819,7 +819,7 @@ int rvu_policy_init(struct rvu *rvu)
 		goto error;
 	}
 
-	max = hw->block[BLKADDR_NIX0].lf.max;
+	max = hw->block[BLKADDR_NIX0].lf.max + hw->block[BLKADDR_NIX1].lf.max;
 	rvu->pf_limits.nix = quotas_alloc(rvu->hw->total_pfs, max, max,
 					  0, &rvu->rsrc_lock, &pf_limit_ops);
 	if (!rvu->pf_limits.nix) {
-- 
2.31.1

